src/theory/evaluator.cpp

```diff
@@ -524,6 +552,460 @@ EvalResult Evaluator::evalInternal(TNode n,
           break;
         }
 
+        case kind::CONST_FLOATINGPOINT:
+        {
+          results[currNode] = EvalResult(currNodeVal.getConst<FloatingPoint>());
+          break;
+        }
+
+        case kind::CONST_ROUNDINGMODE:
+        {
+          results[currNode] = EvalResult(currNodeVal.getConst<RoundingMode>());
+          break;
+        }
+
+        case kind::FLOATINGPOINT_FP:
+        {
+          const BitVector& sign = results[currNode[0]].d_bv;
+          const BitVector& exp = results[currNode[1]].d_bv;
+          const BitVector& sig = results[currNode[2]].d_bv;
+          Assert(sign.getSize() == 1);
+          unsigned e = exp.getSize();
+          unsigned s = sig.getSize() + 1;
+          results[currNode] =
+              EvalResult(FloatingPoint(e, s, sign.concat(exp.concat(sig))));
+          break;
+        }
+
+        case kind::FLOATINGPOINT_EQ:
+        case kind::FLOATINGPOINT_GEQ:
+        case kind::FLOATINGPOINT_GT:
+        {
+          // These kinds should have been removed by earlier rewrites. In
+          // debug, we would like to know about this. In production, we can
+          // just return an invalid result and fall back to the full rewriter.
+          Assert(false);
```

Comment:
This is true as things stand but this does make me a little bit nervous.

Reply:
Whenever we use the evaluator, we have a fallback to the normal rewriter: if the evaluator detects that it can't handle something, it returns a null node and the rewriter can give it a try. That means that this should be safe to do but we also get an alert that something changed when we are running with debug.